src/common/snippets/include/snippets/op/horizon_sum.hpp

```diff
@@ -25,6 +26,7 @@ class HorizonSum : public ov::op::Op {
     bool visit_attributes(AttributeVisitor& visitor) override { return true;}
     std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& new_args) const override;
     void validate_and_infer_types() override;
+    using ShapeInfer = HorizonMax::ShapeInfer;
```

Comment:
I'm not sure that it's good idea to reuse shape infer from another class. I understand that they have the same semantic (as `Broadcast` ops) but if we add new horizon ops, we will always reuse from `HorizonMax`. For me, it's not clear. Is it possible to create common shape infer for all `HorizonOp`s? What do you think?

Reply:
I think that there's no big difference while we have only two ops 😁 
You have a point though, so I moved it to the common part